index.js

```diff
@@ -19,16 +18,7 @@ function sizeof(object) {
       else {
         var bytes = 0;
         _.forOwn(object, function (value, key) {
-          bytes += sizeof(key);
-          try {
-            bytes += sizeof(value);
-          } catch (ex) {
-            if(ex instanceof RangeError) {
-              // circular reference detected, final result might be incorrect
```

Comment:
Using `Set` is a good idea, unfortunately it requires ES6, so it will not work on node 0.12, which is still quite popular. 
Set-like functionality could be implemented manually:
- store every reference in an array, 
- when adding a new one - check if it exists, if yes then do not add
- calculate sizeof for all references

Will put it to roadmap :)